leakcanary-analyzer/src/main/java/com/squareup/leakcanary/LeakTraceElement.kt

```diff
@@ -0,0 +1,143 @@
+   * is excluded. There is always at least one element.
+   */
+  val classHierarchy: List<String>,
+
+  /** Additional information, may be null.  */
+  val extra: String?,
+
+  /** If not null, there was no path that could exclude this element.  */
+  val exclusion: Exclusion?,
+
+  /** List of all fields (member and static) for that object.  */
+  val fieldReferences: List<LeakReference>
+) : Serializable {
+
+  @Deprecated(
+      "Use {@link #reference} and {@link LeakReference#getDisplayName()} instead.\n" +
+          "    Null if this is the last element in the leak trace, ie the leaking object."
+  )
+  val referenceName: String?
+
+  @Deprecated(
+      "Use {@link #reference} and {@link LeakReference#type} instead.\n" +
+          "    Null if this is the last element in the leak trace, ie the leaking object."
+  )
+  val type: Type?
+
+  val className: String
+
+  @Deprecated("Use {@link #fieldReferences} instead.")
+  val fields: List<String>
+
+  enum class Type {
+    INSTANCE_FIELD,
+    STATIC_FIELD,
+    LOCAL,
+    ARRAY_ENTRY
+  }
+
+  enum class Holder {
+    OBJECT,
+    CLASS,
+    THREAD,
+    ARRAY
+  }
+
+  init {
+    this.referenceName = reference?.displayName
+    this.type = reference?.type
+    this.className = classHierarchy[0]
+    val stringFields = mutableListOf<String>()
+    fieldReferences.forEach { leakReference ->
+      stringFields.add(leakReference.toString())
+    }
+    fields = stringFields.toList()
+  }
+
+  /**
+   * Returns the string value of the first field reference that has the provided referenceName, or
+   * null if no field reference with that name was found.
+   */
+  fun getFieldReferenceValue(referenceName: String): String? {
+    return fieldReferences.find { fieldReference -> fieldReference.name.equals(referenceName) }
+        ?.value
+  }
+
+  /** @see [isInstanceOf][] */
+  fun isInstanceOf(expectedClass: Class<out Any>): Boolean {
+    return isInstanceOf(expectedClass.name)
+  }
+
+  /**
+   * Returns true if this element is an instance of the provided class name, false otherwise.
+   */
+  fun isInstanceOf(expectedClassName: String): Boolean {
+    return classHierarchy.find { className -> className.equals(expectedClassName) }
+        .isNullOrEmpty()
+  }
+
+  /**
+   * Returns {@link #className} without the package.
+   */
+  fun getSimpleClassName(): String {
+    val separator = className.lastIndexOf('.')
+    return if (separator == -1) className else className.substring(separator + 1)
+  }
+
+  override fun toString(): String {
+    return toString(false)
+  }
+
+  fun toString(maybeLeakCause: Boolean): String {
+    val staticString = if (reference != null && reference.type == STATIC_FIELD) "static" else ""
+    val holderString =
+      if (holder == ARRAY || holder == THREAD) "${holder.name.toLowerCase(US)} " else ""
+    val simpleClassName = getSimpleClassName()
+    val referenceName = if (reference != null) ".${reference.displayName}" else ""
```

Comment:
You could use the elvis operator on all null checks as another kotlin feature.
https://kotlinlang.org/docs/reference/null-safety.html#elvis-operator